deploy/concierge/deployment.yaml

```diff
@@ -265,3 +285,16 @@ spec:
       loadBalancerIP: #@ data.values.impersonation_proxy_spec.service.load_balancer_ip
       #@ end
       annotations: #@ data.values.impersonation_proxy_spec.service.annotations
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: #@ defaultResourceNameWithSuffix("impersonation-proxy")
+  namespace: #@ namespace()
+  labels: #@ labels()
+  annotations:
+    #! wait until the SA exists to create this secret so that the token controller does not delete it
+    #! we have this secret at the end so that kubectl will create the service account first
+    kapp.k14s.io/change-rule: "upsert after upserting impersonation-proxy.concierge.pinniped.dev/serviceaccount"
```

Comment:
This `impersonation-proxy.concierge.pinniped.dev` string needs to be dynamic based on the API suffix.

Reply:
Uh why?

Reply:
I got confused here. I thought this name referred directly to the name of the ServiceAccount, which is chosen dynamically:

```yaml
name: #@ defaultResourceNameWithSuffix("impersonation-proxy")
```

It's actually the name of the "change group", which is static:
```yaml
    kapp.k14s.io/change-group: "impersonation-proxy.concierge.pinniped.dev/serviceaccount"
```